File: data/c_sharp_skeleton/caching/src/main/java/com/iluwatar/caching/Caching/database/MongoDb.cs
using System;
using MongoDB.Bson;
using MongoDB.Driver;


namespace Com.Iluwatar.Caching
{
    /**
     * Implementation of DatabaseManager.
     * implements base methods to work with MongoDb.
     */
    public class MongoDb : DbManager
    {
        private static readonly string DATABASE_NAME = "admin";
        private static readonly string MONGO_USER = "root";
        private static readonly string MONGO_PASSWORD = "rootpassword";
        private MongoClient client;
        private IMongoDatabase db;

        public void SetDb(IMongoDatabase db)
        {
        }

        /**
         * Connect to Db. Check the connection
         */
        public void Connect()
        {
            db = client.GetDatabase(DATABASE_NAME);
        }

        public void Disconnect()
        {
        }

        /**
         * Read data from DB.
         *
         * @param userId {@link String}
         * @return {@link UserAccount}
         */
        public UserAccount ReadFromDb(string userId)
        {
            return null;
        }

        /**
         * Write data to DB.
         *
         * @param userAccount {@link UserAccount}
         * @return {@link UserAccount}
         */
        public UserAccount WriteToDb(UserAccount userAccount)
        {
            db.GetCollection<BsonDocument>("").InsertOne(null);
            return null;
        }

        /**
         * Update DB.
         *
         * @param userAccount {@link UserAccount}
         * @return {@link UserAccount}
         */
        public UserAccount UpdateDb(UserAccount userAccount)
        {
            db.GetCollection<BsonDocument>("").UpdateOne(null, null);
            return null;
        }

        /**
         * Update data if exists.
         *
         * @param userAccount {@link UserAccount}
         * @return {@link UserAccount}
         */
        public UserAccount UpsertDb(UserAccount userAccount)
        {
            db.GetCollection<BsonDocument>("").UpdateOne(null, null);
            return null;
        }
    }
}

File: data/java_skeleton/caching/src/main/java/com/iluwatar/caching/database/MongoDb.java
/*
 * This project is licensed under the MIT license. Module model-view-viewmodel is using ZK framework licensed under LGPL (see lgpl-3.0.txt).
 *
 * The MIT License
 * Copyright © 2014-2022 Ilkka Seppälä
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package com.iluwatar.caching.database;

import static com.iluwatar.caching.constants.CachingConstants.ADD_INFO;
import static com.iluwatar.caching.constants.CachingConstants.USER_ACCOUNT;
import static com.iluwatar.caching.constants.CachingConstants.USER_ID;
import static com.iluwatar.caching.constants.CachingConstants.USER_NAME;

import com.iluwatar.caching.UserAccount;
import com.iluwatar.caching.constants.CachingConstants;
import com.mongodb.MongoClient;
import com.mongodb.MongoClientOptions;
import com.mongodb.MongoCredential;
import com.mongodb.ServerAddress;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.model.UpdateOptions;
import lombok.extern.slf4j.Slf4j;
import org.bson.Document;

/**
 * Implementation of DatabaseManager.
 * implements base methods to work with MongoDb.
 */
@Slf4j
public class MongoDb implements DbManager {
  private static final String DATABASE_NAME = "admin";
  private static final String MONGO_USER = "root";
  private static final String MONGO_PASSWORD = "rootpassword";
  private MongoClient client;
  private MongoDatabase db;

  void setDb(MongoDatabase db) {
      return;
  }

  /**
   * Connect to Db. Check th connection
   */
  @Override
  public void connect() {
      return;
  }

  @Override
  public void disconnect() {
      return;
  }

  /**
   * Read data from DB.
   *
   * @param userId {@link String}
   * @return {@link UserAccount}
   */
  @Override
  public UserAccount readFromDb(final String userId) {
      return null;
  }

  /**
   * Write data to DB.
   *
   * @param userAccount {@link UserAccount}
   * @return {@link UserAccount}
   */
  @Override
  public UserAccount writeToDb(final UserAccount userAccount) {
      return null;
  }

  /**
   * Update DB.
   *
   * @param userAccount {@link UserAccount}
   * @return {@link UserAccount}
   */
  @Override
  public UserAccount updateDb(final UserAccount userAccount) {
      return null;
  }

  /**
   * Update data if exists.
   *
   * @param userAccount {@link UserAccount}
   * @return {@link UserAccount}
   */
  @Override
  public UserAccount upsertDb(final UserAccount userAccount) {
      return null;
  }
}
